committee_admissions/admissions/views.py

```diff
@@ -48,39 +42,52 @@ def get_serializer_class(self):
         return AdmissionPublicSerializer
 
 
-class CommitteeViewSet(LoginRequiredMixin, viewsets.ModelViewSet):
+class CommitteeViewSet(viewsets.ModelViewSet):
     queryset = Committee.objects.all()
     serializer_class = CommitteeSerializer
+    permission_classes = [CommitteePermissions]
 
 
-class UserApplicationViewSet(viewsets.ModelViewSet):
-    queryset = UserApplication.objects.all()
-    serializer_class = UserApplicationSerializer
-    # authentication_classes = []
-    permission_classes = [permissions.IsAuthenticatedOrReadOnly, IsOwnerOrReadOnly]
-
-
-class ApplicationViewSet(viewsets.ModelViewSet):
-    queryset = UserApplication.objects.all()
+class ApplicationViewSet(LoginRequiredMixin, viewsets.ModelViewSet):
+    queryset = UserApplication.objects.all().select_related("admission", "user")
     serializer_class = ApplicationCreateUpdateSerializer
-    # authentication_classes = []
-    permission_classes = [permissions.IsAuthenticatedOrReadOnly, IsOwnerOrReadOnly]
+    permission_classes = [ApplicationPermissions]
+
+    def get_queryset(self):
+        user = self.request.user
+        if Membership.objects.filter(
+            user=user,
+            role=constants.LEADER,
+            abakus_group__name="Hovedstyret",
+        ).exists():
+            return super().get_queryset().prefetch_related(
+                'committee_applications', 'committee_applications__committee'
+            )
+
+        group = Membership.objects.filter(user=user, role=constants.LEADER).first().abakus_group
+
+        qs = CommitteeApplication.objects.filter(committee__name=group.name
+                                                 ).select_related('committee')
+
+        return super().get_queryset().filter(committee_applications__committee__name=group.name
```

Comment:
This looks like acid, but it does indeed work :100: 

Tests next! :1st_place_medal: